gallery/__init__.py

```diff
@@ -791,7 +797,14 @@ def display_dir_thumbnail(dir_id: int, auth_dict: Optional[Dict[str, Any]] = Non
         thumbnail_uuid = thumbnail_uuid.split('.')[0]
 
     link = storage_interface.get_link("thumbnails/{}".format(thumbnail_uuid))
-    return redirect(link)
+    if "LOCAL_STORAGE_PATH" in app.config:
+        link = "http://" + app.config["SERVER_NAME"] + link
+    req = requests.get(link)
+    if req.status_code == requests.codes.ok:
+        return req.content
+    dir_model.thumbnail_uuid = refresh_directory_thumbnail(dir_model)
```

Comment:
is there not a way to refresh file thumbnails in the file thumbnail route like you're doing here?

Reply:
File Thumbnails really don't have a way to refresh like this, file thumbnails are reliant on the file they refer to. If it can't load the file, it can't load it. If it's a directory, then it depends on the files within it.

Reply:
If a file thumbnail 404s, can we try regenerating it, and if it fails, give the default rather than a 404?